python/ray/serve/_private/common.py

```diff
@@ -35,7 +35,66 @@ def to_replica_actor_class_name(self):
         return f"ServeReplica:{self.app_name}:{self.name}"
 
 
-ReplicaTag = str
+@dataclass(frozen=True)
+class ReplicaID:
+    unique_id: str
+    deployment_id: DeploymentID
+    full_id_str_prefix: str = "SERVE_REPLICA::"
+
+    def to_full_id_str(self) -> str:
```

Comment:
Non-blocker: Can probably just  rename this to something like `full_id` and make it a property